### PR TITLE
fix(FEEL): vars in filters are properly parsed

### PR DESCRIPTION
**Pull request by Skaiir**
Related to https://github.com/camunda/team-hto/issues/279

### CODE REVIEW

File: packages/form-js-viewer/test/spec/util/GetSchemaVariables.spec.js [hunk 1]
```diff
@@ -148,4 +149,16 @@ describe('util/getSchemaVariables', () => {
     ]);
   });
 
+
+  it('should include variables in ships example', () => {
+
+    const variables = getSchemaVariables(shipsExampleSchema);
+
+    // currently produces: ['selectedShip', undefined] :zap:
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/test/spec/util/GetSchemaVariables.spec.js:157]:
```suggestion
```

**Review by pinussilvestrus — COMMENTED** [review]
Really nice 👍 

👍 for all the additional test cases !

**Comment by Skaiir** [comment]
@pinussilvestrus Fixed the one comment and also decided to filter null and undefined variables. Mostly so that we don't ever get that issue again.

File: packages/form-js-viewer/src/util/index.js [hunk 2]
```diff
@@ -143,14 +143,12 @@ export function getSchemaVariables(schema, expressionLanguage = new FeelExpressi
...
       if (property && !expressionLanguage.isExpression(property) && templating.isTemplate(property)) {
-
         const templateVariables = templating.getVariableNames(property);
-
         variables = [ ...variables, ...templateVariables ];
       }
     });
 
-    return variables;
+    return variables.filter(variable => variable !== undefined || variable !== null);
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/src/util/index.js:151]: Not 100% sure about this one. The good thing about the current behavior (keeping null values) was that we catched the error in Tasklist really easily and it didn't slip our attention. However we should not crash the whole workflow, so I believe it's fine now.

**Review by pinussilvestrus — APPROVED** [review]

**Pull request merged**